src_c/base.c

```diff
@@ -2173,62 +2153,45 @@ MODINIT_DEFINE(base)
 #endif
 
     apiobj = encapsulate_api(c_api, "base");
-    if (apiobj == NULL) {
-        Py_XDECREF(atexit_register);
-        Py_DECREF(pgExc_BufferError);
-        Py_DECREF(module);
-        return NULL;
-    }
-    ecode = PyDict_SetItemString(dict, PYGAMEAPI_LOCAL_ENTRY, apiobj);
-    Py_DECREF(apiobj);
-    if (ecode) {
-        Py_XDECREF(atexit_register);
-        Py_DECREF(pgExc_BufferError);
-        Py_DECREF(module);
-        return NULL;
+    if (PyModule_AddObject(module, PYGAMEAPI_LOCAL_ENTRY, apiobj)) {
+        Py_XDECREF(apiobj);
+        goto error;
     }
 
     if (PyModule_AddIntConstant(module, "HAVE_NEWBUF", 1)) {
-        Py_XDECREF(atexit_register);
-        Py_DECREF(pgExc_BufferError);
-        Py_DECREF(module);
-        return NULL;
+        goto error;
     }
 
-    if (!is_loaded) {
-        /*some intialization*/
-        PyObject *quit = PyObject_GetAttrString(module, "quit");
-        PyObject *rval;
+    /*some intialization*/
+    PyObject *quit = PyObject_GetAttrString(module, "quit");
+    PyObject *rval;
 
-        if (quit == NULL) { /* assertion */
-            Py_DECREF(atexit_register);
-            Py_DECREF(pgExc_BufferError);
-            Py_DECREF(module);
-            return NULL;
-        }
-        rval = PyObject_CallFunctionObjArgs(atexit_register, quit, NULL);
-        Py_DECREF(atexit_register);
-        Py_DECREF(quit);
-        if (rval == NULL) {
-            Py_DECREF(module);
-            Py_DECREF(pgExc_BufferError);
-            return NULL;
-        }
-        Py_DECREF(rval);
-        Py_AtExit(pg_atexit_quit);
+    if (!quit) { /* assertion */
+        goto error;
+    }
+    rval = PyObject_CallFunctionObjArgs(atexit_register, quit, NULL);
+    Py_DECREF(atexit_register);
+    Py_DECREF(quit);
+    atexit_register = NULL;
+    if (!rval) {
+        goto error;
+    }
+    Py_DECREF(rval);
+    Py_AtExit(pg_atexit_quit);
 #ifdef HAVE_SIGNAL_H
-        pg_install_parachute();
+    pg_install_parachute();
 #endif
 
 #ifdef MS_WIN32
-        SDL_RegisterApp("pygame", 0, GetModuleHandle(NULL));
-#endif
-#if defined(macintosh)
-#if (!defined(__MWERKS__) && !TARGET_API_MAC_CARBON)
-        SDL_InitQuickDraw(&pg_qd);
-#endif
+    SDL_RegisterApp("pygame", 0, GetModuleHandle(NULL));
+#elif defined(macintosh) && !defined(__MWERKS__) && !TARGET_API_MAC_CARBON
+    SDL_InitQuickDraw(&pg_qd);
```

Comment:
QuickDraw is obsolete, so this can be removed.

Reply:
aight, good idea